graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java

```diff
@@ -185,7 +238,9 @@ private void recordTimestamp(List<Map.Entry<IndexSet, Message>> messageList, Set
 
     private BulkResult runBulkRequest(final Bulk request, int count) {
         try {
-            return BULK_REQUEST_RETRYER.call(() -> client.execute(request));
+            // Enable Expect-Continue to catch 413 errors before we send the actual data
+            final RequestConfig requestConfig = RequestConfig.custom().setExpectContinueEnabled(true).build();
```

Comment:
I noticed the following in the documentation for `RequestConfig#isExpectContinueEnabled()`:

> 'Expect: 100-continue' handshake should be used with caution, as it may cause problems with HTTP servers and proxies that do not support.

So I wonder if we should make this configurable in `graylog.conf`? (default: enabled) Otherwise we cannot do anything if a user has a broken proxy in front of Elasticsearch.

Reply:
good idea. I've added an option